From 2591d0189257caeaae2057dfed0a260310497a61 Mon Sep 17 00:00:00 2001
From: Reuben Hawkins <reubenhwk@gmail.com>
Date: Tue, 4 Oct 2011 13:18:47 -0700
Subject: [PATCH] removing mdelay in unicast only case

---
 process.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

Index: radvd-1.8/process.c
===================================================================
--- radvd-1.8.orig/process.c	2011-10-14 23:17:31.000000000 +0200
+++ radvd-1.8/process.c	2011-10-14 23:17:51.659622373 +0200
@@ -195,7 +195,6 @@
 
 	if (iface->UnicastOnly) {
 		dlog(LOG_DEBUG, 3, "random mdelay for %s: %g seconds.", iface->Name, delay/1000.0);
-		mdelay(delay);
 		send_ra_forall(iface, &addr->sin6_addr);
 	}
 	else if ( timevaldiff(&tv, &iface->last_multicast) / 1000.0 < iface->MinDelayBetweenRAs ) {
