From 2591d0189257caeaae2057dfed0a260310497a61 Mon Sep 17 00:00:00 2001
From: Reuben Hawkins <reubenhwk@gmail.com>
Date: Tue, 4 Oct 2011 13:18:47 -0700
Subject: [PATCH] removing mdelay in unicast only case

---
 process.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/process.c b/process.c
index 393a753..cac6fdc 100644
--- a/process.c
+++ b/process.c
@@ -191,7 +191,6 @@ process_rs(int sock, struct Interface *iface, unsigned char *msg, int len,
 	dlog(LOG_DEBUG, 3, "random mdelay for %s: %.2f", iface->Name, delay);
  	
 	if (iface->UnicastOnly) {
-		mdelay(delay);
 		send_ra_forall(sock, iface, &addr->sin6_addr);
 	}
 	else if ((tv.tv_sec + tv.tv_usec / 1000000.0) - (iface->last_multicast_sec +
-- 
1.7.6.3

